tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -10806,6 +10806,9 @@ void hlsl::HandleDeclAttributeForHLSL(Sema &S, Decl *D, const AttributeList &A,
   case AttributeList::AT_HLSLGroupShared:
     declAttr = ::new (S.Context) HLSLGroupSharedAttr(A.getRange(), S.Context,
       A.getAttributeSpellingListIndex());
+    if (VarDecl *VD = dyn_cast<VarDecl>(D)) {
```

Comment:
What happens if it's not a VarDecl? If it can't happen, then remove the if and use non-dyn `cast`

Reply:
I think it could happen in an invalid use, where we will emit an error, but also don't want to crash here.